src/rust/src/metrics_ffi.rs

```diff
@@ -64,18 +56,30 @@ pub extern "C" fn metrics_run(
 
     allow_ips
         .into_iter()
-        .fold(
-            PrometheusBuilder::new().listen_address(bind_address),
-            |builder, subnet| builder.add_allowed(subnet),
+        .try_fold(
+            PrometheusBuilder::new().with_http_listener(bind_address),
+            |builder, subnet| {
+                builder.add_allowed_address(subnet).map_err(|e| {
+                    error!("Invalid -metricsallowip argument '{}': {}", subnet, e);
+                    e
+                })
+            },
         )
-        .install()
+        .and_then(|builder| builder.install())
+        .map_err(|e| {
+            error!("Could not install Prometheus exporter: {}", e);
+            e
+        })
```

Comment:
In the case where a `-metricsallowip` argument is invalid, we'll log twice. This should be:
```suggestion
        .and_then(|builder| builder.install()
            .map_err(|e| {
                error!("Could not install Prometheus exporter: {}", e);
                e
            })
        )
```